File: 2223/2223.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumOfScoresOfBuiltStrings = function(root) {
    if (!root) return 0;
    return 1 + Math.max(sumOfScoresOfBuiltStrings(root.left), sumOfScoresOfBuiltStrings(root.right));
};

File: 2223/2223.go
func sumOfScoresOfBuiltStrings(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := sumOfScoresOfBuiltStrings(root.Left)
	r := sumOfScoresOfBuiltStrings(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}